fix(typescript/prefer-namespace-keyword): skip nested `TSModuleDeclaration`s

overlookmotel opened the pull request:
Prevent fix replacing text within the module name or block in the case of nested modules.

Before this PR:

Input:

```ts
module X.Y.module {
  x = "module";
}
```

After fix:

```ts
namespace X.Y.namespace {
  x = "namespace";
}
```

Prevent this by exiting early if the `TSModuleDeclaration` is nested (i.e. `Y` or `module` in above example).

Note: I discovered this bug when I replaced early `return` with `unwrap` in #15805. IMO it's better to `unwrap` when you believe the check cannot fail, as the panic may alert us to bugs which may otherwise lie undiscovered.

Side note: The design we have with nested `TSModuleDeclaration`s representing `module X.Y.Z {}` is pretty unintutive. I'd like to change it if we can.

overlookmotel commented:
It's really hard to imagine all the possible ways these kinds of searches can go wrong just by thinking about it. Maybe it'd be better to reduce the slice of text that gets searched. e.g. in this case, only search source text between the start of the `TSModuleDeclaration` and the start of its `id`. That would have caught this.

camc314 reviewed (approved):
I got worried that:
```
declare module X.Y.module { x = 'module'; }
declare module X.Y { declare module module { x = 'module'; } }
``

would have the same ast for a second 🙂

overlookmotel committed:
fix(typescript/prefer-namespace-keyword): skip nested `TSModuleDeclaration`s (#15806)

Prevent fix replacing text within the module name or block in the case of nested modules.

Before this PR:

Input:

```ts
module X.Y.module {
  x = "module";
}
```

After fix:

```ts
namespace X.Y.namespace {
  x = "namespace";
}
```

Prevent this by exiting early if the `TSModuleDeclaration` is nested (i.e. `Y` or `module` in above example).

Note: I discovered this bug when I replaced early `return` with `unwrap` in #15805. IMO it's better to `unwrap` when you believe the check cannot fail, as the panic may alert us to bugs which may otherwise lie undiscovered.

Side note: The design we have with nested `TSModuleDeclaration`s representing `module X.Y.Z {}` is pretty unintutive. I'd like to change it if we can.
